src/arrayMethodJoin.js

```diff
@@ -4,8 +4,33 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let str = '';
+    let sep = separator;
+
+    if (this.length === 0) {
+      return str;
+    }
+
+    if (sep === null) {
+      sep = `${separator}`;
+    }
```

Comment:
You can use concatenation inside your loop instead of if condition. For instance: str += '' + sep;